artifactory/cli_test.go

```
@@ -30,3 +41,135 @@ func TestValidateGoNativeCommand(t *testing.T) {
+		expectedProps   string
+		expectedPattern string
+		expectedBuild   string
+		expectedBundle  string
+		expectError     bool
+	}{
+		{"withoutPattern", []string{"key1=val1"}, []string{}, "key1=val1", "", "", "", true},
+		{"withPattern", []string{"TestPattern", "key1=val1"}, []string{}, "key1=val1", "TestPattern", "", "", false},
+		{"withBuild", []string{"key1=val1"}, []string{"build=buildName/buildNumber"}, "key1=val1", "*", "buildName/buildNumber", "", false},
+		{"withBundle", []string{"key1=val1"}, []string{"bundle=bundleName/bundleVersion"}, "key1=val1", "*", "", "bundleName/bundleVersion", false},
+		{"withSpec", []string{"key1=val1"}, []string{"spec=" + getSpecPath(tests.SetDeletePropsSpec)}, "key1=val1", "${REPO1}/", "", "", false},
+		{"withSpecAndPattern", []string{"TestPattern", "key1=val1"}, []string{"spec=" + getSpecPath(tests.SetDeletePropsSpec)}, "key1=val1", "", "", "", true},
+		{"withPatternAndBuild", []string{"TestPattern", "key1=val1"}, []string{"build=buildName/buildNumber"}, "key1=val1", "TestPattern", "buildName/buildNumber", "", false},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			context, buffer := createContext(test.flags, test.args)
+			propsCommand, err := preparePropsCmd(context)
+			if test.expectError {
+				assert.Error(t, err, buffer)
+			} else {
+				assert.NoError(t, err, buffer)
+				assert.Equal(t, test.expectedProps, propsCommand.Props())
+				assert.Equal(t, test.expectedPattern, propsCommand.Spec().Get(0).Pattern)
+			}
+		})
+	}
+}
+
+func createContext(testFlags, testArgs []string) (*cli.Context, *bytes.Buffer) {
+	flagSet := flag.NewFlagSet("TestFlagSet", flag.ContinueOnError)
+	flags := append(setStringFlags(flagSet, testFlags...), testArgs...)
+	flagSet.Parse(flags)
+
+	buffer := &bytes.Buffer{}
+	app := cli.NewApp()
+	app.Writer = &bytes.Buffer{}
+	return cli.NewContext(app, flagSet, nil), buffer
+}
+
+func getSpecPath(spec string) string {
+	dir, _ := os.Getwd()
```

Let's not ignore the error returned by the ```os.Getwd()``` function.